src/utils.c

```diff
@@ -7,6 +7,10 @@ bool within_int32_repres(double x) {
   return R_FINITE(x) && x < 2147483648 && x > -2147483648;
 }
 
+bool within_int64_repres(double x) {
+  return R_FINITE(x) && x < (double)INT64_MAX && x > (double)INT64_MIN;
```

Comment:
< and not <= ?

Reply:
The boundary also needs to be `INT64_MIN+1` since `INT64_MIN = NA_INTEGER64`

Reply:
Nice catch! Though I'm not sure it matters, those numbers are not exactly represented in type `double` anyway right?